src/main/java/com/redhat/devtools/lsp4ij/usages/LSPUsageSearcher.java

```diff
@@ -103,7 +106,19 @@ public void processElementUsages(@NotNull PsiElement element, @NotNull Processor
                     // Show response of textDocument/definition, textDocument/references, etc as usage info.
                     List<LSPUsagePsiElement> usages = usagesFuture.getNow(null);
                     if (usages != null) {
-                        for (LSPUsagePsiElement usage : usages) {
+                        // Remove any usages that fully contain other usages, e.g., definitions when what's really
+                        // wanted is the contained declaration/name identifier
+                        List<LSPUsagePsiElement> filteredUsages = new ArrayList<>(usages);

```

Comment:
One thing that has bugged me for quite a while is that some language servers will return a usage for the _definition_ of a declaration that represents its entire text range and another usage for its _declaration_ that represents its name identifier text range. The latter is what the user ultimately cares about in a Find Usages scenario, though. This filtering ensures that when one usage fully contains one or more other usages, it is filtered out so only the most specific usages are presented to the end user.